product_contract/models/sale_order.py

```diff
@@ -84,8 +82,13 @@ def action_create_contract(self):
                     raise ValidationError(
                         _(
                             "You must specify a contract "
-                            "template for '{}' product in '{}' company."
-                        ).format(order_line.product_id.name, rec.company_id.name)
+                            "template for '%(product_name)s' product "
+                            "in '%(company_name)s' company."
+                        )
```

Comment:
Prefer using translation method attributes.

`_("You must specify (...) %(product_name)s (...)", product_name=(...), company_name=(...))`